src/components/ui4/SplitRowsView.tsx

```diff
@@ -5,26 +5,33 @@ import { cacheStyles } from 'react-native-patina'
 import { Theme, useTheme } from '../services/ThemeContext'
 
 interface Props {
-  children: Array<{ left: React.ReactNode; right: React.ReactNode }>
+  children: SplitRow[] | SplitRow
+}
+
+interface SplitRow {
+  left: React.ReactNode
+  right: React.ReactNode
 }
 
 /**
  * A view representing rows of data split on the left and right edges of the
  * line. Neither side will exceed 50% of the width of the view.
  **/
-export const SplitRowView = (props: Props) => {
+export const SplitRowsView = (props: Props) => {
   const { children } = props
   const theme = useTheme()
   const styles = getStyles(theme)
 
+  const getSplitRow = (row: SplitRow, key?: string) => (
+    <View key={key} style={styles.row}>
+      <View style={styles.leftColumn}>{row.left}</View>
+      <View style={styles.rightColumn}>{row.right}</View>
+    </View>
+  )
+
   return (
     <View style={styles.container}>
-      {children.map((row, index) => (
-        <View key={index} style={styles.row}>
-          <View style={styles.leftColumn}>{row.left}</View>
-          <View style={styles.rightColumn}>{row.right}</View>
-        </View>
-      ))}
+      {Array.isArray(children) ? children.map((row: SplitRow, index: number) => getSplitRow(row, `${index}-${row.left}`)) : getSplitRow(children)}
```

Comment:
Is it valid to put a React node into `${}`? It seems like the index alone would be a unique key.